encoding/resampling.md

```diff
@@ -1,35 +1,581 @@
+Therefore,
+they only interpolate “missing” values,
+leaving the input samples untouched.
+
+This is true for filters that evaluate to 0
+at all integer positions except 0
+or whose support is <= 1.
+Examples include:
+(Windowed) Sinc filters,
+such as Lanczos,
+Bicubic with B=0,
+e.g. Hermite and Catmull-Rom,
+and Triangle/Bilinear.
+
+This can be a beneficial property in some cases,
+for example the No-Op case.
+No-Op means that no scaling, shifting or similiar is performed,
+that is, the input is resampled at exactly the same positions.
+In this case,
+an interpolation filter will return the input image untouched,
+whereas other filters will slightly alter it.
+
+Another,
+more practical,
+case where this property is useful
+is when shifting an image by full pixel widths (integers),
+again because input pixel values aren’t changed
+but just relocated.
+
+### Two-dimensional resampling
+
+There are two different ways to go about
+resampling in two dimensions.
+
+#### Tensor resampling (also: orthogonal, 2-pass, seperated)
+
+The image is resampled in two seperate passes:
+First it is resampled horizontally, then vertically.
+This allows images to be treated 1-dimensionally
+since each pixel row/column can be resampled seperately.
+The main advantage of this method is that it's extremly fast,
+which is why it’s the much more common one;
+generally, unless indicated otherwise,
+this is what is used.
+
+#### EWA (Elliptical Weighted Averaging) (also: cylindrical, polar, circular)
+
+![Two-dimensional kernel. The radius is colored green.](images/polar.png)
+
+All input samples whose [Euclidian distance][L2] to the pixel
+is within the filter’s radius contribute to its value.
+The Euclidian distance is passed to the filter kernel.
+This is a lot more costly than tensor resampling in terms of runtime.
+
+[L2]: https://en.wikipedia.org/wiki/Euclidean_distance
+
+### Scaling in modified colorspaces
+
+The colorspace used when resampling
+can significantly impact the output’s subjective quality.
+
+#### Downscaling in linear light
+
+Downscaling in gamma-corrected light instead of linear light
+can sometimes noticeably dim the image.
+To see why this happens,
+consider this gradient:
+
+![A grayscale gradient from 0 to 255.](images/gamma_shades.jpg)
+
+It is should be apparent
+that the brightness doesn’t scale linearly with the pixel values.
+This is because most digital video
+uses [gamma-transformed][gamma] pixel values
+in order to compress more perceptually distinct color shades
+into 8 bit.
+This causes the encoded pixel values
+to deviate from their expected brightness,
+e.g. a grey pixel has value 187 instead of 127 in sRGB.
+This poses a problem when merging and interpolating colors,
+because the average pixel value of two colors no longer corresponds
+to their average perceived brightness.
+For example,
+if you wanted to merge black and white (0 and 255),
+you would expect to get grey,
+but since grey actually has a value of ca. 187,
```

Comment:
@Lypheo is this an intended `ca.` here? If so, can you expand on what it stands for. If not I'll remove it.

Reply:
That’s supposed to be a circa, yes. Seems to be sort of a false friend, sorry (in German, we use it synonymously with “approximately”). You can remove it, I suppose.

Reply:
Changed to a `~` in 19052ed9ce4b547fad33d3d1339fe08bfaffe97b